soc/raspberrypi/rpi_pico/Kconfig.soc

```diff
@@ -0,0 +1,25 @@
+# Raspberry Pi (RP) MCU line
+
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# Copyright (c) 2024 Andrew Featherstone
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_RPI_PICO
+	bool
+
+config SOC_FAMILY
+	default "rpi_pico" if SOC_FAMILY_RPI_PICO
+
+config SOC_SERIES
+	default "rp2040" if SOC_SERIES_RP2040
+	default "rp2350" if SOC_SERIES_RP2350
+
+config SOC_SERIES_RP2040
+	bool
+	select SOC_FAMILY_RPI_PICO
+
+config SOC_SERIES_RP2350
+	bool
+	select SOC_FAMILY_RPI_PICO
```

Comment:
these should go in the Kconfig.soc files in the sub folders

Reply:
Nordic is an exception in this case in order to support the `bsim` targets and emulated hw. Instead, the preferred way is to add the series in the underlying `Kconfig.soc` files under the folders, as in for example https://github.com/zephyrproject-rtos/zephyr/tree/main/soc/nxp/imx

Reply:
I've raised #83007 so that such "exception"s are documented, corrected, or both.